wholesale/entity.py

```diff
@@ -0,0 +1,90 @@
+from common.constants import entity_key, cache_id
+from entity.abstract import ResourceEntity
+from datetime import date, datetime
+
+import re
+import json
+import requests
+
+API_URL = 'https://data.cityofnewyork.us/resource/87fx-28ei.json' + \
+    '?$select=`market`,`bic_number`,`account_name`,`application_type`,`postcode`' + \
+    '&$where=disposition_date >= \'{}\' and disposition_date <= \'{}\' &$limit=10000&$offset={}' + '&$order=disposition_date'
+
+ZIPCODE_PATTERN = re.compile('^\\d{5}$')
+ALPHA_ONLY_PATTERN = re.compile('^[A-Za-z\\s]+$')
+
+class WholesaleMarket(ResourceEntity):
+
+    @staticmethod
+    def dependencies():
+        return[
+            entity_key.census_us_zipcode,
+            entity_key.calendar_date,
+            entity_key.wholesale_market_app_type
+        ]
+
+    def get_zipcode_id(self, record, field):
+        zipcode_entity = self.dependencies_map[entity_key.census_us_zipcode]
+        zipcode = zipcode_entity.get_cached_value(record[field])
+        return zipcode['id'] if zipcode else None
+    
+    def get_market_app_type_id(self, record, field):
+        app_type_entity = self.dependencies_map[entity_key.wholesale_market_app_type]
+        app_type = app_type_entity.get_cached_value(record[field])
+        return app_type['id'] if app_type else None
+
+    def __init__(self):
+        super().__init__()
+
+        self.table_name = 'wholesale_market'
+        self.fields = [
+            {'field': 'market'},
+            {'field': 'account_name'},
+            {'field': 'application_type', 'column': 'market_application_type_id', 'data': self.get_market_app_type_id},
+            {'field': 'postcode', 'column': 'zipcode_id', 'data': self.get_zipcode_id}
+        ]
+        self.cacheable_fields = ['account_name']
+    
+    def skip_record(self, record):
+        response = False
+
+        if self.record_cache:
+            for key, value in self.record_cache.items():
```

Comment:
We need to be careful with this as it's `O(n^2)` complexity. You can create a your own cache by overriding the [load_cache](https://github.com/Project-REFOCUS/dashboard-core-dataset-application/blob/main/entity/abstract.py#L48) method on the parent class.

Then you can ensure all values within the cache are already lowercase.

Reply:
I thought about this and should have asked